service/Core/DataFormats/Office/MsWordDecoder.cs

```diff
@@ -4,27 +4,49 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
 using DocumentFormat.OpenXml.Packaging;
 using DocumentFormat.OpenXml.Wordprocessing;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.Diagnostics;
+using Microsoft.KernelMemory.Pipeline;
 
 namespace Microsoft.KernelMemory.DataFormats.Office;
 
-public class MsWordDecoder
+public class MsWordDecoder : IContentDecoder
 {
-    public FileContent ExtractContent(string filename)
+    private readonly ILogger<MsWordDecoder> _log;
+
+    public IEnumerable<string> SupportedMimeTypes { get; } = new[] { MimeTypes.MsWordX, MimeTypes.MsWord };
```

Comment:
ditto

Reply:
Same as above, if you're agree I can remove all the checks for the old Office file formats and rely on the default skip message.